core/unit_test/TestResize.hpp

```diff
@@ -132,8 +132,339 @@ void testResize() {
+                      sizes[4], sizes[5], sizes[6]);
+    typename view_type::HostMirror h_view_7d_old =
+        Kokkos::create_mirror(view_7d);
+    Kokkos::deep_copy(view_7d, 777);
+    Kokkos::deep_copy(h_view_7d_old, view_7d);
+    Kokkos::resize(Kokkos::view_resize(Kokkos::WithoutInitializing), view_7d,
+                   2 * sizes[0], sizes[1], sizes[2], sizes[3], sizes[4],
+                   sizes[5], sizes[6]);
+    EXPECT_TRUE(view_7d.extent(0) == 2 * sizes[0]);
+    typename view_type::HostMirror h_view_7d =
+        Kokkos::create_mirror_view(view_7d);
+    Kokkos::deep_copy(h_view_7d, view_7d);
+    bool test = true;
+    for (size_t i0 = 0; i0 < sizes[0]; ++i0) {
+      for (size_t i1 = 0; i1 < sizes[1]; ++i1) {
+        for (size_t i2 = 0; i2 < sizes[2]; ++i2) {
+          for (size_t i3 = 0; i3 < sizes[3]; ++i3) {
+            for (size_t i4 = 0; i4 < sizes[4]; ++i4) {
+              for (size_t i5 = 0; i5 < sizes[5]; ++i5) {
+                for (size_t i6 = 0; i6 < sizes[6]; ++i6) {
+                  if (h_view_7d(i0, i1, i2, i3, i4, i5, i6) !=
+                      h_view_7d_old(i0, i1, i2, i3, i4, i5, i6)) {
+                    test = false;
+                    break;
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+    EXPECT_TRUE(test == true);
+  }
+  {
+    typedef Kokkos::View<int********, DeviceType> view_type;
+    view_type view_8d("view_8d", sizes[0], sizes[1], sizes[2], sizes[3],
+                      sizes[4], sizes[5], sizes[6], sizes[7]);
+    typename view_type::HostMirror h_view_8d_old =
+        Kokkos::create_mirror(view_8d);
+    Kokkos::deep_copy(view_8d, 888);
+    Kokkos::deep_copy(h_view_8d_old, view_8d);
+    Kokkos::resize(Kokkos::view_resize(Kokkos::WithoutInitializing), view_8d,
+                   2 * sizes[0], sizes[1], sizes[2], sizes[3], sizes[4],
+                   sizes[5], sizes[6], sizes[7]);
+    EXPECT_TRUE(view_8d.extent(0) == 2 * sizes[0]);
+    typename view_type::HostMirror h_view_8d =
+        Kokkos::create_mirror_view(view_8d);
+    Kokkos::deep_copy(h_view_8d, view_8d);
+    bool test = true;
+    for (size_t i0 = 0; i0 < sizes[0]; ++i0) {
+      for (size_t i1 = 0; i1 < sizes[1]; ++i1) {
+        for (size_t i2 = 0; i2 < sizes[2]; ++i2) {
+          for (size_t i3 = 0; i3 < sizes[3]; ++i3) {
+            for (size_t i4 = 0; i4 < sizes[4]; ++i4) {
+              for (size_t i5 = 0; i5 < sizes[5]; ++i5) {
+                for (size_t i6 = 0; i6 < sizes[6]; ++i6) {
+                  for (size_t i7 = 0; i7 < sizes[7]; ++i7) {
+                    if (h_view_8d(i0, i1, i2, i3, i4, i5, i6, i7) !=
+                        h_view_8d_old(i0, i1, i2, i3, i4, i5, i6, i7)) {
+                      test = false;
+                      break;
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+    EXPECT_TRUE(test == true);
+  }
```

Comment:
This block isn't even though it is something we should have been testing for as well in the default case with data initialization.

Reply:
@dalg24, thanks. The default case is tested in my new commit.